sound/soc/intel/boards/sof_es8336.c

```diff
@@ -119,8 +97,7 @@ static void pcm_pop_work_events(struct work_struct *work)
 
 	gpiod_set_value_cansleep(priv->gpio_speakers, priv->speaker_en);
 
-	if (quirk & SOF_ES8336_HEADPHONE_GPIO)
-		gpiod_set_value_cansleep(priv->gpio_headphone, priv->speaker_en);
+	gpiod_set_value_cansleep(priv->gpio_headphone, !priv->speaker_en);
```

Comment:
No (at least on Huawei Matebook D15): there, the headphone GPIO is reversed: true disables, false enables.
So, the right code is, actually:

```gpiod_set_value_cansleep(priv->gpio_headphone, priv->speaker_en);```

as before.